### PR TITLE
Load usage listeners from bundles

### PR DESCRIPTION
**Pull request by sjcorbett**
And JcloudsLocation's image choosers.

### CODE REVIEW

File: core/src/test/java/org/apache/brooklyn/core/mgmt/internal/LocalUsageManagerTest.java [hunk 1]
```diff
@@ -16,67 +16,77 @@
...
     @Test
     public void testAddUsageListenerViaProperties() throws Exception {
         BrooklynProperties brooklynProperties = BrooklynProperties.Factory.newEmpty();
         brooklynProperties.put(UsageManager.USAGE_LISTENERS, RecordingStaticUsageListener.class.getName());
         mgmt = LocalManagementContextForTests.newInstance(brooklynProperties);
-        
+        assertUsageListenerCalledWhenApplicationStarted();
+    }
+
+    @Test(expectedExceptions = ClassCastException   .class)
```
Comment (Graeme-Miller) [inline core/src/test/java/org/apache/brooklyn/core/mgmt/internal/LocalUsageManagerTest.java]: what's with the space between ClassCastException and .class

File: core/src/main/java/org/apache/brooklyn/core/mgmt/internal/LocalUsageManager.java [hunk 2]
```diff
@@ -173,20 +180,23 @@
...
-                if (listener instanceof UsageListener) {
-                    addUsageListener((UsageListener)listener);
-                } else if (listener == null) {
-                    throw new NullPointerException("null listener in config "+UsageManager.USAGE_LISTENERS);
+            for (Object obj : listeners) {
+                if (obj == null) {
+                    throw new NullPointerException("null listener in config " + UsageManager.USAGE_LISTENERS);
+                } else if (!(obj instanceof UsageListener)) {
+                    throw new ClassCastException("Configured object is not a UsageListener. This probably means coercion failed: " + obj);
                 } else {
```
Comment (Graeme-Miller) [inline core/src/main/java/org/apache/brooklyn/core/mgmt/internal/LocalUsageManager.java]: doesn't really need to be an else anymore

File: core/src/test/java/org/apache/brooklyn/core/mgmt/internal/LocalUsageManagerTest.java (hunk 1 above)
Comment (sjcorbett) [inline core/src/test/java/org/apache/brooklyn/core/mgmt/internal/LocalUsageManagerTest.java]: Accidental tab. Why doesn't GitHub show I've fixed this any more?

**Comment by neykov** [comment]
LGTM, thanks for fixing `JCloudsLocation` as well. Merging.

**Pull request merged**